### PR TITLE
add median feature to frontend jax numpy statistical

### LINKED ISSUE #12589: median

### PR DESCRIPTION
**Pull request by trantuantdt**
Close #12589

### CODE REVIEW

**Comment by trantuantdt** [comment]
Hi @nassimberrada . There are some errors. But they come from features that others develop. Could you please review my pull request?

**Comment by trantuantdt** [comment]
Hi @Ishticode . Could you please review my pull request?

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py [hunk 1]
```diff
@@ -729,3 +729,40 @@ def test_jax_numpy_std(
...
+# median
+@handle_frontend_test(
+    fn_tree="jax.numpy.median",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
+        min_value=-2**10,
+        max_value=2**10,
+        valid_axis=True,
+    ),
+    keemdims=st.booleans(),
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:743]: this should be `keepdims`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py [hunk 2]
```diff
@@ -729,3 +729,40 @@ def test_jax_numpy_std(
         atol=1e-3,
         rtol=1e-3,
     )
+
+
+# median
+@handle_frontend_test(
+    fn_tree="jax.numpy.median",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:738]: the function jax.numpy.median supports more types besides just float. We should try to be as comprehensive as possible. Please try "valid" here to include all valid dtypes for the tests.

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @trantuantdt Thanks for the PR. Sorry for the delay but it was not linked to an issue. I will try linking the PR with the issue you mentioned. You PR is great. Some changes are requested. Please feel free to question further or request a review when you think it's ready.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py (hunk 2 above)
Comment (trantuantdt) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_statistical.py:738]: Thank @Ishticode . When I change to "valid" I get an error "the ground truth framework jax returned a float32 datatype while the backend numpy returned a float64 datatype". But the data input is uint8. I have stuck with this error. Do you have any recomemd for me? What should I try? 
Screenshot from 2023-03-27 17-09-11

**Comment by trantuantdt** [comment]
Hi @Ishticode . Could you please review my PR. I solved the problem "the ground truth framework jax returned a float32 datatype while the backend numpy returned a float64 datatype" by converted backend dtype of result to desired dtype. There were also some errors about the shape of result on backend torch. So I adjusted function "quantile" to return correct shape of the result.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by trantuantdt** [comment]
I'm waiting for review

**Review by Ishticode — CHANGES_REQUESTED** [review]
@trantuantdt 
Firstly, apologies for the late review on this PR. I will keep it on my radar for high priority. Thank you for the patience. Mostly looking good. I slightly updated the specification of supported dtypes in tensorflow implementation. Although I when I checked the tfp percentile function does work for interger inputs also so I am not entirely sure we need to specify float as the (only) supported dtype as the `with_supported_dtype` decorator implies.

In an ideal situation we wonldn't need to modify both backend and frontend in a single PR. But this should be fine hopefully.
I will try to get this moving soon if you could in to the issue I mentioned above bcz removing the the decorator to only support floats does lead to test failure.
Thanks 🙂 @trantuantdt

**Comment by trantuantdt** [comment]
Hi @Ishticode ,tfp.stats.percentile linear and midpoint interpolation do not work with integer dtypes in tensorflow document

**Review by Ishticode — APPROVED** [review]
Yes you are right! my bad for overlooking that. Thank you very much for the contribution @trantuantdt. I will get this merged. 🙂
